drivers/flash/nrf_qspi_nor.c

```diff
@@ -476,17 +476,42 @@ static inline int qspi_nor_read_id(struct device *dev,
 	return 0;
 }
 
+#if (CONFIG_HEAP_MEM_POOL_SIZE > 0)
+u8_t *rx_ram_buf;
+u8_t *tx_ram_buf;
+#endif
+
 static int qspi_nor_read(struct device *dev, off_t addr, void *dest,
 			 size_t size)
 {
+	int ret;
+
 	if (!dest) {
 		return -EINVAL;
 	}
+#if (CONFIG_HEAP_MEM_POOL_SIZE > 0)
+	rx_ram_buf = NULL;
+	void *tmp_dest = dest;
+	size_t var_size = size;
+
+	if (((uint32_t)dest < CONFIG_SRAM_BASE_ADDRESS)
+	   || (size < sizeof(uint32_t))) {
+		if (size < sizeof(uint32_t)) {
+			size = 4;
+		}
+		rx_ram_buf = k_malloc(size);
```

Comment:
dosen't malloc for 4 B is a overkill here?
Why no to use a variable on stack instead?

Reply:
I was unsure if the stack lived throughout the read. As I assumed it was DMA based. If it does then sure